0.10.0 breaks a mock that was working with 0.9.1

Hi, just after updating to 0.10.0 I got a failure for the following mock, which was compiling fine before:

```rust
  mockall::mock! {
    TestTransactionReader {}
    impl TransactionsReader for TestTransactionReader {
      fn read_transactions<'a>(&'a mut self) -> Box<dyn Stream<Item = Result<Transaction>> + Unpin + 'a>;
    }
  }
```

The error that I get says:

```
error[E0759]: `self` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement
  --> src/processors/simple.rs:72:3
   |
72 | /   mockall::mock! {
73 | |     TestTransactionReader {}
74 | |     impl TransactionsReader for TestTransactionReader {
75 | |       fn read_transactions<'a>(
...  |
78 | |     }
79 | |   }
   | |   ^
   | |   |
   | |___this data with an anonymous lifetime `'_`...
   |     ...is captured and required to live as long as `'static` here
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Let me know if there is anything else I should report to help you with this ;-)
Great work. Best

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,14 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [ Unreleased ] - ReleaseDate
+
+## Fixed
+
+- Fix mocking trait methods whose return values have lifetime parameters, a
+  regression since v0.10.0.
+  ([#304](https://github.com/asomers/mockall/pull/304))
+
 ## [ 0.10.0 ] - 2021-06-27
 
 ### Added

--- a/mockall/tests/mock_generic_method_returning_nonstatic.rs
+++ b/mockall/tests/mock_generic_method_returning_nonstatic.rs
@@ -10,6 +10,11 @@ use mockall::*;
 #[derive(Clone)]
 struct X<'a>(&'a u32);
 
+trait Bong {
+    fn trait_foo<'a>(&self) -> X<'a>;
+    fn trait_baz<'a>(&self) -> &X<'a>;
+}
+
 mock! {
     Thing {
         fn foo<'a>(&self) -> X<'a>;
@@ -28,6 +33,11 @@ mock! {
         // Mockall requires generic methods' generic parameters to implement
         // std::any::Any, which means they must be 'static.
         //fn bean<'a, T: 'static>(&self, t: T) -> X<'a>;
+    }
+    // The same types of methods should work if they are Trait methods.
+    impl Bong for Thing {
+        fn trait_foo<'a>(&self) -> X<'a>;
+        fn trait_baz<'a>(&self) -> &X<'a>;
     }
 }
 
@@ -66,4 +76,42 @@ fn return_nonstatic() {
         .returning(move || xstatic.clone());
 
     assert_eq!(42u32, *thing.foo().0);
+}
+
+mod trait_methods {
+    use super::*;
+
+    #[test]
+    fn return_static() {
+        const D: u32 = 42;
+        let x = X(&D);
+        let mut thing = MockThing::new();
+        thing.expect_trait_foo()
+            .return_const(x);
+
+        assert_eq!(42u32, *thing.trait_foo().0);
+    }
+
+    #[test]
+    fn return_static_ref() {
+        const D: u32 = 42;
+        let x = X(&D);
+        let mut thing = MockThing::new();
+        thing.expect_trait_baz()
+            .return_const(x);
+
+        assert_eq!(42u32, *(*thing.trait_baz()).0);
+    }
+
+    #[test]
+    fn return_nonstatic() {
+        let d = 42u32;
+        let x = X(&d);
+        let xstatic: X<'static> = unsafe{ std::mem::transmute(x) };
+        let mut thing = MockThing::new();
+        thing.expect_trait_foo()
+            .returning(move || xstatic.clone());
+
+        assert_eq!(42u32, *thing.trait_foo().0);
+    }
 }

--- a/mockall_derive/src/mock_function.rs
+++ b/mockall_derive/src/mock_function.rs
@@ -619,10 +619,10 @@ impl MockFunction {
         -> impl ToTokens
     {
         let inner_mod_ident = self.inner_mod_ident();
-        if let Some(sa) = self_args {
+        if let Some(PathArguments::AngleBracketed(abga)) = self_args {
             assert!(!self.is_method_generic(),
                 "specific impls with generic methods are TODO");
-            quote!(#inner_mod_ident::Expectation #sa)
+            quote!(#inner_mod_ident::Expectation #abga)
         } else {
             // staticize any lifetimes.  This is necessary for methods that
             // return non-static types, because the Expectation itself must be

